h2o-core/src/main/java/hex/faulttolerance/Recovery.java

```diff
@@ -13,8 +19,30 @@
 import java.util.*;
 
 public class Recovery<T extends Keyed> {
+    
+    private static final Logger LOG = Logger.getLogger(Recovery.class);
 
     public static final String REFERENCES_META_FILE_SUFFIX = "_references";
+    public static final String RECOVERY_META_FILE = "recovery.json";
+    
+    public static final String INFO_CLASS = "class";
+    public static final String INFO_RESULT_KEY = "resultKey";
+    public static final String INFO_JOB_KEY = "jobKey";
+
+    public static void autoRecover(String autoRecoveryDir) {
```

Comment:
A perfect indicator for a `Recovery` interface and `FileSystemAutoRecovery` implementation - others might come. Strategy pattern. Or could be refactored in future once a new strategy is added. Just noticed this obvious clue.

Reply:
sounds like premature refactoring